staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -157,7 +172,46 @@ func (s *SpdyRoundTripper) dial(req *http.Request) (net.Conn, error) {
 
 	rwc, _ := proxyClientConn.Hijack()
 
-	if req.URL.Scheme != "https" {
+	return s.tlsConn(req.URL, rwc, targetHost)
+}
+
+// dialWithSocks5Proxy dials the host specified by url through a socks5 proxy.
+func (s *SpdyRoundTripper) dialWithSocks5Proxy(req *http.Request, proxyURL *url.URL) (net.Conn, error) {
+	// ensure we use a canonical host with proxyReq
+	targetHost := netutil.CanonicalAddr(req.URL)
+	proxyDialAddr := netutil.CanonicalAddr(proxyURL)
+
+	var auth *proxy.Auth
+	if proxyURL.User != nil {
+		pass, _ := proxyURL.User.Password()
+		auth = &proxy.Auth{
+			User:     proxyURL.User.Username(),
+			Password: pass,
+		}
+	}
+	proxyDialer, err := proxy.SOCKS5("tcp", proxyDialAddr, auth, proxy.Direct)
+
+	if err != nil {
+		return nil, err
+	}
+
+	proxyDialConn, err := proxyDialer.Dial("tcp", targetHost)
```

Comment:
I don't see us passing a context into here or into `proxy.SOCKS5`... what ensures this dial is cancelled if the parent context times out or is cancelled?

Reply:
it looks like the value returned from `proxy.SOCKS5` implements `ContextDialer` ... should we expect/use that and plumb in the `req.Context()`?

Reply:
Solved:

`proxyDialConn, err := proxyDialer.(proxy.ContextDialer).DialContext(req.Context(), "tcp", targetHost)`